lib/phoenix_live_view/channel.ex

```diff
@@ -255,6 +255,17 @@ defmodule Phoenix.LiveView.Channel do
     handle_redirect(state, command, flash, nil)
   end
 
+  def handle_info({:phoenix_live_reload, _topic, _changed_file}, %{socket: socket} = state) do
+    Phoenix.CodeReloader.reload(socket.endpoint)
+
+    new_socket =
+      Enum.reduce(socket.assigns, socket, fn {key, val}, socket ->
+        Utils.force_assign(socket, key, val)
+      end)
+
+    handle_changed(state, new_socket, nil)
```

Comment:
This is tricky because if the user changed `mount` or `handle_params`, those wouldn't be picked up. I partially feel the best way to go around this is to send a client a refresh and the client performs a full remount. Probably doing the changes in #2326.